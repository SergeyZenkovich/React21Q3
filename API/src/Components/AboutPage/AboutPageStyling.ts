import React from "react";
import styled from "styled-components";

import fontPBold from "../../assets/fonts/MavenPro-Bold.ttf";

export const StyledAboutPage = styled.div`
  @font-face {
    font-family: "CardTextFontBold";
    src: url(${fontPBold});
  }
  display: flex;
  justify-content: center;
  & h3 {
    font-family: "CardTextFontBold";
  }
`;
