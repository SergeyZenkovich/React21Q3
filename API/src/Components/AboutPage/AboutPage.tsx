import React, {useEffect} from "react";
import {motion, AnimatePresence} from "framer-motion";
import styled from "styled-components";
import fontPBold from "../../assets/fonts/MavenPro-Bold.ttf";

import {StyledAboutPage} from "./AboutPageStyling";
import aboutImg from "../../assets/img/info.png";

const AboutPage = ({
  setActivePage
}: {
  setActivePage: React.Dispatch<React.SetStateAction<string>>;
}): JSX.Element => {
  useEffect(() => {
    setActivePage("About");
  }, [setActivePage]);
  return (
    <StyledAboutPage
      as={motion.div}
      initial={{opacity: 0}}
      animate={{opacity: 1}}
      exit={{opacity: 0}}
    >
      <h3>This is AboutPage</h3>
      <img src={aboutImg} alt="" />
    </StyledAboutPage>
  );
};

export default AboutPage;
