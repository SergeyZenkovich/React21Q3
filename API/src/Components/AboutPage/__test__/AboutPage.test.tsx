import React from "react";
import ReactDOM from "react-dom";
import {act} from 'react-dom/test-utils';
import AboutPage from "../AboutPage";

it("renders without crashing", () => {
  const container = document.createElement("div");
act(()=>{
  ReactDOM.render(<AboutPage setActivePage={() => {}}></AboutPage>, container);
});
  const header = container.querySelector("h3");
  expect(header.textContent).toBe("This is AboutPag");
});
