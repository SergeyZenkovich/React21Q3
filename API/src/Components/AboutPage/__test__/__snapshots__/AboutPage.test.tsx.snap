// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AboutPage Component matches shapshot 1`] = `
<div
  className="sc-bdnxRM juTZgK"
  style={
    Object {
      "opacity": 0,
    }
  }
>
  <h3>
    This is AboutPage
  </h3>
  <img
    alt=""
    src="test-file-stub"
  />
</div>
`;
